perun-core/src/main/java/cz/metacentrum/perun/core/entry/MembersManagerEntry.java

```diff
@@ -128,6 +130,27 @@ public Member createSpecificMember(PerunSession sess, Vo vo, Candidate candidate
 		return getMembersManagerBl().createSpecificMember(sess, vo, candidate, specificUserOwners, specificUserType, groups);
 	}
 
+	public Member createSponsoredAccount(PerunSession sess, Map<String, String> params, String namespace, ExtSource extSource, String extSourcePostfix, Vo vo) throws InternalErrorException, PrivilegeException, UserNotExistsException, ExtSourceNotExistsException, UserExtSourceNotExistsException, WrongReferenceAttributeValueException, ExtendMembershipException, AlreadyMemberException, WrongAttributeValueException, GroupOperationsException {
+		Utils.checkPerunSession(sess);
+		Utils.notNull(extSource, "extSource");
+		Utils.notNull(namespace, "namespace");
+		Utils.notNull(vo, "vo");
+		Utils.notNull(extSourcePostfix, "extSourcePostfix");
+
+		if (!AuthzResolver.isAuthorized(sess, Role.VOADMIN, vo)) {
```

Comment:
We should probably allow also REGISTRAR user role, since it will be called by external application, but maybe we can make their account to be the VO manager. It depends in which VO we will need to create sponsors in. I'm letting this to @licehammer to decide.